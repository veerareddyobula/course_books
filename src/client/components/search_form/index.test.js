import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import SearchForm from './index';

const mockStore = configureStore([]);

describe('SearchForm Component', () => {
  let store;
  let component;
  beforeEach(() => {
    store = mockStore({
      fetchTitlesList: () => true,
      booksTitlesList: []
    });
    const onSubmitHandler = () => true;
    store.dispatch = jest.fn();
    component = renderer.create(
      <Provider store={store}>
        <SearchForm submitHandler={onSubmitHandler} />
      </Provider>
    );
  });
  it('SearchForm Component Snapshot', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
