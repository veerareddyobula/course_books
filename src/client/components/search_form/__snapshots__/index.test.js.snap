// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm Component SearchForm Component Snapshot 1`] = `
<form
  action="#"
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="search-form"
  >
    <div
      className="auto-complete"
    >
      <input
        name="queryString"
        onBlur={[Function]}
        onChange={[Function]}
        value=""
      />
      
    </div>
    <div>
      <button
        className="btn-primary"
        disabled={true}
        type="submit"
      >
        Submit
      </button>
    </div>
  </div>
</form>
`;
